packages/common/src/TransactionUtils.ts

```diff
@@ -101,17 +121,32 @@ export const runTransaction = async ({
     const maximumGasPriceMultiple = 2 * 3;
     // Pre-London transactions require `gasPrice`, London transactions require `maxFeePerGas` and `maxPriorityFeePerGas`
 
-    let receipt;
+    let receipt: any;
 
     // If the config contains maxPriorityFeePerGas then this is a London transaction. In this case, simply use the
     // provided config settings but double the maxFeePerGas to ensure the transaction is included, even if the base fee
     // spikes up. The difference between the realized base fee and maxFeePerGas is refunded in a London transaction.
     if (transactionConfig.maxFeePerGas && transactionConfig.maxPriorityFeePerGas) {
-      receipt = await transaction.send({
-        ...transactionConfig,
-        maxFeePerGas: parseInt(transactionConfig.maxFeePerGas.toString()) * 2,
-        type: "0x2",
-      } as SendOptions);
+      if (waitForMine)
+        receipt = await transaction.send({
```

Comment:
I know we've already discussed this a bit, but would the following work?

```ts
        receiptPromiEvent = transaction.send({
          ...transactionConfig,
          maxFeePerGas: parseInt(transactionConfig.maxFeePerGas.toString()) * 2,
          type: "0x2",
        } as SendOptions);
        hash = await new Promise((resolve) => {
         receiptPromiEvent.on("transactionHash", (hash) => {
              resolve({ receipt: { transactionHash: hash } as any, transactionConfig, returnValue: null as any });
         });
...
return { receipt: (receipt as unknown) as PromiEvent<TransactionReceipt>, hash, returnValue, transactionConfig };

```

This approach would allow us to drop the `waitForMine` argument and would mean a little less casework, but it would mean that we _always_ return and unresolved PromiEvent for the receipt that the caller would then need to await on, which might add some messiness to the existing bots.

Just wanted to throw an alternative idea out, but I think what you have here works well and might be slightly easier on the existing callers.

Reply:
I really like this as we can store all the promises and then at the end of the execution block use a `Promose.all` to wait for all transactions to mine. If any of them revert/error out then we can log accordingly.

Reply:
please see the refined implementation.